gdnative-sys/build.rs

```diff
@@ -76,304 +77,304 @@ mod header_binding {
+    use std::convert::AsRef;
+    use std::fs::File;
+    use std::io::Write as _;
+    use std::path;
 
     #[derive(Debug, serde::Deserialize)]
-    struct CoreApi<'a> {
-        #[serde(rename = "type")]
-        type_: &'a str,
-        version: Version,
-        #[serde(borrow)]
-        next: Option<Box<CoreApi<'a>>>,
-        #[serde(borrow)]
-        api: Vec<ApiFunction<'a>>,
-    }
-
-    impl<'a> CoreApi<'a> {
-        fn struct_type_name(&self, root: bool) -> String {
-            if root {
-                "godot_gdnative_core_api_struct".to_string()
-            } else {
-                format!(
-                    "godot_gdnative_core_{}_{}_api_struct",
-                    self.version.major, self.version.minor
-                )
-            }
-        }
+    struct ApiRoot {
+        core: Api,
+        extensions: Vec<Api>,
     }
 
     #[derive(Debug, serde::Deserialize)]
-    struct ExtensionApi<'a> {
-        name: Option<&'a str>,
+    struct Api {
+        name: Option<String>,
         #[serde(rename = "type")]
-        type_: &'a str,
+        type_: String,
         version: Version,
-        #[serde(borrow)]
-        next: Option<Box<ExtensionApi<'a>>>,
-        #[serde(borrow)]
-        api: Vec<ApiFunction<'a>>,
-    }
-
-    impl<'a> ExtensionApi<'a> {
-        fn struct_type_name(&self, root: bool) -> String {
-            if root {
-                format!(
-                    "godot_gdnative_ext_{}_api_struct",
-                    self.type_.to_lowercase()
-                )
-            } else {
-                format!(
-                    "godot_gdnative_ext_{}_{}_{}_api_struct",
-                    self.type_.to_lowercase(),
-                    self.version.major,
-                    self.version.minor
-                )
-            }
-        }
+        next: Option<Box<Api>>,
+        #[serde(rename = "api")]
+        functions: Vec<Function>,
     }
 
     #[derive(Debug, serde::Deserialize)]
     struct Version {
-        major: i64,
-        minor: i64,
+        major: u32,
+        minor: u32,
     }
 
-    type ArgumentType<'a> = &'a str;
-    type ArgumentName<'a> = &'a str;
+    #[derive(Debug, serde::Deserialize)]
+    struct Function {
+        name: String,
+        return_type: String,
+        arguments: Vec<Argument>,
+    }
 
     #[derive(Debug, serde::Deserialize)]
-    struct ApiFunction<'a> {
-        name: &'a str,
-        return_type: &'a str,
-        #[serde(borrow)]
-        arguments: Vec<(ArgumentType<'a>, ArgumentName<'a>)>,
+    struct Argument {
+        type_: String,
+        name: String,
     }
 
-    pub(crate) fn generate(manifest_dir: &str, out_dir: &str) {
-        let api_json_path = format!("{}/godot_headers/gdnative_api.json", manifest_dir);
+    impl ApiRoot {
+        fn all_apis(&self) -> Vec<&Api> {
+            let mut result = Vec::new();
+            result.extend(self.core.nexts());
+            for extension in &self.extensions {
```

Comment:
Previously, the Android extension was skipped because "it uses types from headers we are not binding to", in `assemble_ext_def`. Something similar isn't seen in your code. Have you confirmed that the Android extension works now?

Reply:
I have not, I merely wrapped what was created by bindgen.

Reply:
I think the original comment referred to `JNIEnv` and `jobject`, found in `godot_android.h`. The current version now redefines those types as opaque pointers with conditional compilation, so I think there's no need for action here.